### ISSUE TITLE
[a11y]: Inline CodeSnippet content is not read by JAWS

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#13671]
### Package

@carbon/react

### Browser

Firefox

### Operating System

Windows

### Package version

1.27.0

### React version

v18

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

[Reported internally](https://ibm-studios.slack.com/archives/C054WK4ES5S/p1682510942552969), when testing the inline code snippet with JAWS, 

> the screen reader does not identify the code snippet that is to be copied. Hence the Blind user would not know what content is to be copied
>
> Note: in Vo iPhone when the focus reaches, , SR announces "copy to clip board" button. After activation, the label changes to "Node-v". this confuses the user.
>
> SR should identify the code snippet along with the "code" role so that the content is programmatically identified

One additional recommended potential course of action included separating the content from the button and providing an `aria-describedby`. We could also review this with the Carbon Accessibility Guild to determine some other ways we could potentially tackle this.

It's possible this is related to or was introduced with https://github.com/carbon-design-system/carbon/pull/13460

### WCAG 2.1 Violation

1.3.1 Info and Relationships 

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-codesnippet--inline

### Steps to reproduce

Load the story and use JAWS to see how the content is not read
